### PR TITLE
test: Add VM cache test

### LINKED ISSUE #1259: Need test for vmcache function
https://github.com/kata-containers/runtime/pull/1166

### PR DESCRIPTION
**Pull request by GabyCT**
This will add a VM cache test, where we will enable this
feature and then we will run a container.

Fixes #1259

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

File: integration/vm_cache/vm_cache_test.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+disable_vm_cache_config() {
+	echo "disable vm cache config"
+	sudo sed -i 's/vm_cache_number = 1/#vm_cache_number = 0/g' "${RUNTIME_CONFIG_PATH}"
+}
+
+init_vm_template() {
+	echo "init vm template"
+	sudo -E PATH=$PATH "$kata_runtime_bin" factory init
```
Comment (ganeshmaharaj) [inline integration/vm_cache/vm_cache_test.sh]: Is running factory init twice intentional? If so, do we not need to check the return code of the first one?

**Comment by ganeshmaharaj** [comment]
Ouch! sorry @GabyCT reviewed the changes and then noticed the WIP label. Please ignore my comments if you are still working on this.

File: integration/vm_cache/vm_cache_test.sh (hunk 1 above)
Comment (GabyCT) [reply integration/vm_cache/vm_cache_test.sh]: @ganeshmaharaj not worries, thanks for the feedback

**Comment by GabyCT** [comment]
@teawater this is kind of the approach that I will take on this test, any feedback? thanks

File: integration/vm_cache/vm_cache_test.sh [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	sudo sed -i 's/vm_cache_number = 1/#vm_cache_number = 0/g' "${RUNTIME_CONFIG_PATH}"
+	sudo sed -i -e 's/^\(use_vsock\).*=.*$/#\1 = true/g' "${RUNTIME_CONFIG_PATH}"
+}
+
+init_vm_template() {
+	sudo sed -i -e 's/^#enable_template.*$/enable_template = true/g' "${RUNTIME_CONFIG_PATH}"
+	echo "init vm template"
+	sudo -E PATH=$PATH "$kata_runtime_bin" factory init &
+	pid=$!
+	sudo kill -9 $pid
```
Comment (amshinde) [inline integration/vm_cache/vm_cache_test.sh]: Why a forced kill here?

File: integration/vm_cache/vm_cache_test.sh [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	clean_env
+	extract_kata_env
+	enable_vm_cache_config
+	init_vm_template
+}
+
+test_docker_vm_cache() {
+	docker run -d --runtime $RUNTIME $IMAGE $PAYLOAD_ARGS
+	check_initrd=$(ps aux | grep ${HYPERVISOR_PATH} | grep initrd)
+	[ ! -z "${check_initrd}" ] || die "vm cache not running properly"
```
Comment (amshinde) [inline integration/vm_cache/vm_cache_test.sh]: Are you planning to add some further checks here. From what I see, we are just checking if the hypervisor is running with initrd, that does not really verify much against vm-cache, unless I missed something.

**Pull request closed without merging**